contracts/src/token/memecoin.cairo

```diff
@@ -408,7 +409,7 @@ mod UnruggableMemecoin {
                 i += 1;
             };
             self.team_allocation.write(team_allocation);
-            self.pre_launch_holders_count.write(initial_holders.len().try_into().unwrap());
+            self.pre_launch_holders_count.write(unique_count(initial_holders).try_into().unwrap());
```

Comment:
@enitrat do you think this'll be less compute expensive than using a dictionary to track if an address appears twice? If an address appears twice, it should just revert because addresses are supposed to be unique

Reply:
@credence0x Considering that the actual number of addresses is at most 10, the computing costs are negligible using a dict / an array. 

I actually used a dict initially but had to revert back to an array as snfoundry had a bug that prevented the test when using a dict. see 3c9650d

Reply:
> If an address appears twice, it should just revert because addresses are supposed to be unique

Regarding this, I don't think it matters if the pre_launch_holders_count indeed counts unique holders.

Reply:
> > If an address appears twice, it should just revert because addresses are supposed to be unique
> 
> Regarding this, I don't think it matters if the pre_launch_holders_count indeed counts unique holders.

in the grand scheme of things, i don't think it matters much either. but the `enforce_prelaunc_holders_limit` function tries to enforce unique holders, so i just think it should be unique. for consistency at least.